proxy.php

```diff
@@ -301,34 +301,73 @@ protected function apply_filters() {
 				],
 			];
 		}
+
+		/**
+		 * If there's no aggregations in the template or if the relation isn't 'and', we are done.
+		 */
+		if ( empty( $this->query['aggs'] ) || 'or' === $this->relation ) {
+			return;
+		}
+
+		/**
+		 * Apply filters to aggregations.
+		 *
+		 * Note the usage of `&agg` (passing by reference.)
+		 */
+		foreach ( $this->query['aggs'] as $agg_name => &$agg ) {
+			$new_filters = [];
+
+			/**
+			 * Only filter an aggregation if there's sub-aggregations.
+			 */
+			if ( empty( $agg['aggs'] ) ) {
+				continue;
+			}
+
+			/**
+			 * Get any existing filter, or a placeholder.
+			 */
+			$existing_filter = $agg['filter'] ?? [ 'match_all' => [ 'boost' => 1 ] ];
+
+			/**
+			 * Get new filters for this aggregation.
+			 *
+			 * Don't apply a filter to a matching aggregation if the relation is 'or'.
+			 */
+			foreach ( $this->filters as $filter_name => $filter ) {
+				if ( $filter_name === $agg_name && 'or' === $this->relation ) {
```

Comment:
is this necessary `'or' === $this->relation`? Since we do the same check on line 308.